### ISSUE TITLE
extern_instances are not generated

### ISSUE THREAD

**Issue by Pfeifenjoy** [p4lang/p4c#1926]
Hey,

maybe I am missing something here but somehow when I compile
file with externs the extern instances are missing.

They are describe here.
https://github.com/p4lang/behavioral-model/blob/master/docs/JSON_format.md#extern_instances

My command is:
```
	p4c-bm2-ss \
		--std p4-16 \
		--p4runtime-files ${BUILD_DIR}/p4c/basic.txt \
		--emit-externs \
		--target bmv2 \
		--arch v1model \
		-o ${BUILD_DIR}/p4c/basic.json \
		p4/switch.p4
```

In my p4 file I instantiate the extern as follows.

```
extern myextern {
	...
}

myextern() myExtern;  // instantiation
```

When I manually add the extern to the extern_instances everything works.
However I would like them to be generated.

**Comment by antoninbas** [comment]
Duplicate of https://github.com/p4lang/p4c/issues/1882

**Comment by mbudiu-vmw** [comment]
Please test the fix for #1882 and let us know if it solves the problem.
